### DISCUSSION TITLE
"error":"invalid_request","error_description":"grant_type is required"

### DISCUSSION THREAD

**Discussion by yusufcole** [etsy/open-api#1090] [Q&A] [answered]
i am trying to get an authorization code from https://www.etsy.com/oauth/connect.
pinging to the app in etsy works, requesting access also works but to get an acces token the request gets stuck with response http code 401....

![Grantacces2](https://github.com/etsy/open-api/assets/7366435/b9d367be-fa41-49bc-b8c0-81a4ee0f4f9d)

the request content look like this....

![PostBody1](https://github.com/etsy/open-api/assets/7366435/78276cc7-8052-49d4-bf19-6ebe4f9fb37b)

but klicking on grantacces always returns 

![ResponseToken](https://github.com/etsy/open-api/assets/7366435/454751da-2dc5-4463-a6bf-d7df27087a7a)

![Response](https://github.com/etsy/open-api/assets/7366435/7edd8930-dd6d-4fe2-aa18-ec851fcbafe2)

**Does anybody know what i am doing wrong please?**

**Comment by emergingdzns** [comment, marked as answer]
I'm not familiar with your code language but that would seem to me as if maybe it's being sent as a GET instead of a POST? Are you posting that JSON string? or are you posting those variables as individual form params? They should be form parameters (not a single json string).

**Reply by yusufcole** [reply]
They should be form parameters (not a single json string).... thanks
